src/EditorFeatures/TestUtilities/Squiggles/AbstractSquiggleProducerTests.cs

```diff
@@ -62,7 +62,7 @@ internal async Task<IList<ITagSpan<IErrorTag>>> GetErrorsFromUpdateSource(TestWo
                 {
                     source.RaiseDiagnosticsUpdated(updateArgs);
 
-                    await wrapper.WaitForTags();
+                    await wrapper.WaitForTags(willBlockOnCompletion: true);
```

Comment:
everything i've seen so far seems to be passing in 'true' here.  Could you give an intuition on who should *not* pass in true?

Reply:
If for some reason you want to wait on items without changing runtime behavior. The thing I didn't like is something that looks like a simple observer producing behavior changes.

Reply:
> If for some reason you want to wait on items without changing runtime behavior. The thing I didn't like is something that looks like a simple observer producing behavior changes. I wasn't especially happy with the design choice so alternative solutions would be welcome.

I think i'd prefer to just change the name here to something that makes that more clear.  i.e. roll this into the name of hte method.

Reply:
Renamed the main wait method and removed the `willBlockOnCompletion` parameters.